library/src/test/java/org/pollub/library/BookServiceTest.java

```diff
@@ -0,0 +1,107 @@
+package org.pollub.library;
+
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.persistence.PersistenceException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.pollub.library.item.model.Book;
+import org.pollub.library.item.model.dto.BookCreateDto;
+import org.pollub.library.item.repository.IBookRepository;
+import org.pollub.library.item.service.BookService;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+class BookServiceTest {
+
+    @Mock
+    private IBookRepository bookRepository;
+
+    @InjectMocks
+    private BookService bookService;
+
+    private BookCreateDto bookCreateDto;
+    private Book book;
+
+    @BeforeEach
+    void setUp() {
+        bookCreateDto = new BookCreateDto();
+        bookCreateDto.setTitle("Test Book");
+        bookCreateDto.setAuthor("Test Author");
+        bookCreateDto.setPageCount(200);
+        bookCreateDto.setPaperType("Paperback");
+        bookCreateDto.setPublisher("Test Publisher");
+        bookCreateDto.setShelfNumber(1);
+        bookCreateDto.setGenre("Fiction");
+        bookCreateDto.setIsbn("123-456-789");
+        bookCreateDto.setDescription("Test Description");
+
+        book = new Book();
+        book.setId(1L);
+        book.setTitle("Test Book");
+    }
+
+    @Test
+    @DisplayName("Ensure correct book creation with given BookCreateDto")
```

Comment:
So the book in preps is diffrent than a book you are trying to create, so its not ensured that correct book is createad, something is not okay here :C